math/vc_soa/include/algebra/math/impl/vc_soa_transform3.hpp

```diff
@@ -0,0 +1,386 @@
+  using value_type = Vc::Vector<scalar_t>;
+
+  /// 3-element "vector" type (does not observe translations)
+  using vector3 = storage::vector<3, value_type, array_t>;
+  /// Point in 3D space (does observe translations)
+  using point3 = vector3;
+  /// Point in 2D space
+  using point2 = storage::vector<2, value_type, array_t>;
+
+  /// 4x4 matrix type
+  using matrix44 = internal::matrix44<scalar_type, array_t>;
+
+  /// Function (object) used for accessing a matrix element
+  using element_getter = internal::element_getter<scalar_type, array_t>;
+
+  /// @}
+
+  /// @name Data objects
+  /// @{
+
+  matrix44 _data;
+  matrix44 _data_inv;
+
+  /// @}
+
+  /// Default constructor: identity
+  ALGEBRA_HOST_DEVICE
+  transform3() = default;
+
+  /// Contructor with arguments: t, x, y, z
+  ///
+  /// @param t the translation (or origin of the new frame)
+  /// @param x the x axis of the new frame
+  /// @param y the y axis of the new frame
+  /// @param z the z axis of the new frame, normal vector for planes
+  ALGEBRA_HOST_DEVICE
+  transform3(const vector3 &t, const vector3 &x, const vector3 &y,
+             const vector3 &z, [[maybe_unused]] bool get_inverse = true)
+      : _data{x, y, z, t}, _data_inv{invert(_data)} {}
+
+  /// Contructor with arguments: t, z, x
+  ///
+  /// @param t the translation (or origin of the new frame)
+  /// @param z the z axis of the new frame, normal vector for planes
+  /// @param x the x axis of the new frame
+  ///
+  /// @note y will be constructed by cross product
+  ALGEBRA_HOST_DEVICE
+  transform3(const vector3 &t, const vector3 &z, const vector3 &x,
+             bool get_inverse = true)
+      : transform3(t, x, cross(z, x), z, get_inverse) {}
+
+  /// Constructor with arguments: translation
+  ///
+  /// @param t is the transform
+  ALGEBRA_HOST_DEVICE
+  transform3(const vector3 &t) : _data{t}, _data_inv{invert(_data)} {}
+
+  /// Constructor with arguments: matrix
+  ///
+  /// @param m is the full 4x4 matrix with simd-vector elements
+  ALGEBRA_HOST_DEVICE
+  transform3(const matrix44 &m) : _data{m}, _data_inv{invert(_data)} {}
+
+  /// Defaults
+  transform3(const transform3 &rhs) = default;
+  ~transform3() = default;
+
+  /// Equality operator
+  ALGEBRA_HOST_DEVICE
+  inline constexpr bool operator==(const transform3 &rhs) const {
+    return (_data == rhs._data);
+  }
+
+  /// Matrix access operator
+  ALGEBRA_HOST_DEVICE
+  inline const value_type &operator()(std::size_t i, std::size_t j) const {
+    return element_getter{}(_data, i, j);
+  }
+  ALGEBRA_HOST_DEVICE
+  inline value_type &operator()(std::size_t i, std::size_t j) {
+    return element_getter{}(_data, i, j);
+  }
+
+  /// The determinant of a 4x4 matrix
+  ///
+  /// @param m is the matrix
+  ///
+  /// @return a sacalar determinant - no checking done
+  ALGEBRA_HOST_DEVICE
+  inline constexpr value_type determinant(const matrix44 &m) const {
+    return m.z[0] * m.y[1] * m.x[2] + m.y[0] * m.z[1] * m.x[2] +
```

Comment:
I think I took the implementation from the other Vc plugin and then removed the terms that are always zero or one, because the transform3 is in homogeneous coordinates (```x[3]```, ```y[3]```, ```z[3]``` are zero and ```t[3]``` is one)

Reply:
I think the determinant of transform3 is:
       [x0 y0 z0 t0]          [ x0 y0 z0 ]
det  [x1 y1 z1 t1] = det [ x1 y1 z1 ] = x0 [ y1 * z2 - z1 y2 ] - y0 [ x1 * z2 - z1 * x2 ] + z0 [ x1 * y2 - y1 * x2 ]
       [x2 y2 z2 t2]          [ x2 y2 z2 ]
       [0  0   0    1]

Reply:
Ah, I see what happened, I forgot to copy paste the minus sign of the first term, because its written in the line above. Thanks!

Reply:
Odd that the unittest passes...

Reply:
x_2 is set to zero in the soa test.

Reply:
Added a minus sign